Web.md

```diff
@@ -133,6 +133,107 @@ A MIME type mismatch, a response whose
 [rejected](http://tc39.github.io/ecma262/#sec-rejectpromise) with a
 `TypeError`.
 
+### Structured serialization of a `WebAssembly.Memory`
+
+A `WebAssembly.Memory` is a
+[serializable object](https://html.spec.whatwg.org/multipage/infrastructure.html#serializable-objects), and will require changes to the
+[HTML Standard](https://html.spec.whatwg.org/multipage/infrastructure.html#serialization-steps) as follows:
+
+[StructuredSerializeInternal](https://html.spec.whatwg.org/multipage/infrastructure.html#structuredserializeinternal) adds a new step after step 12:
+
+* Otherwise, if *value* has a [[WebAssembly.Memory]] internal slot, then:
+  1. Let *size* be *value*.[[BufferObject]].[[ArrayBufferByteLength]].
+  2. Let *maximum* be *value*.[[WebAssembly.Memory]].[[maximum]].
+  3. If *value*.[[BufferObject]] is a Shared Data Block, then:
+     1. If *forStorage* is true, then throw a
+        "[`DataCloneError`](https://heycam.github.io/webidl/#datacloneerror)"
+        [`DOMException`](https://heycam.github.io/webidl/#dfn-DOMException).
+     2. Otherwise,
+        set *serialized* to { [[Type]]: "WebAssembly.Memory",
+        [[ArrayBufferData]]: *value*.[[BufferObject]].[[ArrayBufferData]],
+        [[ArrayBufferByteLength]]: *size*,
+        [[MemoryMaximum]]: *maximum*,
+        [[AgentCluster]]: the [current Realm Record](https://tc39.github.io/ecma262/#current-realm)'s corresponding [agent cluster](https://tc39.github.io/ecma262/#sec-agent-clusters) }.
+  4. Otherwise:
+     1. Let *dataCopy* be [CreateByteDataBlock](https://tc39.github.io/ecma262/#sec-createbytedatablock)(*size*).
+        NOTE: This can throw a RangeError exception on allocation failure.
+     2. Perform [CopyDataBlockBytes](https://tc39.github.io/ecma262/#sec-copydatablockbytes)
+        (*dataCopy*, 0,
+         *value*.[[BufferObject]].[[ArrayBufferData]], 0, *size*).
+     3. Set *serialized* to { [[Type]]: "WebAssembly.Memory",
+        [[ArrayBufferData]]: *dataCopy*, [[ArrayBufferByteLength]]: *size*,
+        [[MemoryMaximum]]: *maximum* }.
+
+[StructuredDeserialize](https://html.spec.whatwg.org/multipage/infrastructure.html#structureddeserialize) adds a new step after step 12:
+
+* Otherwise, if *serialized*.[[Type]] is "WebAssembly.Memory", then:
+  1. If ! [IsSharedArrayBuffer](https://tc39.github.io/ecma262/#sec-issharedarraybuffer)(*serialized*.[[ArrayBufferData]]) is true:
+     1. If *targetRealm*'s corresponding [agent cluster](https://tc39.github.io/ecma262/#sec-agent-clusters) is not
+        *serialized*.[[AgentCluster]], then throw a
+        "[`DataCloneError`](https://heycam.github.io/webidl/#datacloneerror)"
+        [`DOMException`](https://heycam.github.io/webidl/#dfn-DOMException).
+     2. Set *value* to a new WebAssembly.Memory object in *targetRealm*
+        whose [[BufferObject]] internal slot is set to a new
+        [[ArrayBufferData]] internal slot *value* is
```

Comment:
It looks like "a new [[ArrayBufferData]] internal slot *value* is set to a new" should be deleted; this sentence is a bit jumbled.